cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java

```diff
@@ -228,4 +230,16 @@ public void error(Throwable cause) {
     }
     return controller;
   }
+
+  @Nullable
+  private String getAdapterName(Arguments arguments) {
+    // TODO: Currently this logic is super ugly, should try and pass adapter name as its own separate arg
+    if (arguments.hasOption(ProgramOptionConstants.SCHEDULE_NAME)) {
+      String scheduleName = arguments.getOption(ProgramOptionConstants.SCHEDULE_NAME);
+      if (scheduleName.contains(".")) {
```

Comment:
this won't be correct, since it is valid for a schedule name to have a dot. I see there is a todo, is that going to be fixed in a separate PR?